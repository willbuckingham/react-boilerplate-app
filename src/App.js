import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom'

import Home from './pages/Home'
import NotFound from './pages/NotFound'
import './pages/Home/Home.css'

export class App extends React.Component {

  render() {
    return (
      <Router>
        <Switch>
          <Route exact path='/'>
            <Home />
          </Route>
          <Route path="*">
            <NotFound />
          </Route>
        </Switch>
      </Router>
    )
  }
}

export default App
