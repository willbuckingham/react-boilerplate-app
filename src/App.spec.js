import React from 'react'
import { render } from '@testing-library/react'
import { shallow, mount } from 'enzyme'
import App from './App'

describe('app', () => {
  xit('renders learn react link', () => {
    const { getByText } = render(<App />)
    const linkElement = getByText(/learn react/i)
    expect(linkElement).toBeInTheDocument()
  })

  it('shallows', () => {
    const wrapper = shallow(<App />)

    console.log(wrapper.debug())

    expect(wrapper).toMatchSnapshot()
  })

  it('mounts', () => {
    const wrapper = mount(<App />)

    console.log(wrapper.debug())

    expect(wrapper).toMatchSnapshot()
  })
})
