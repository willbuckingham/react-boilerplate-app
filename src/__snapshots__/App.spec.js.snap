// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app mounts 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            key="/"
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <Home>
              <div
                className="App"
              >
                <header
                  className="App-header"
                >
                  <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
                  />
                </header>
              </div>
            </Home>
          </Route>
        </Switch>
      </Router>
    </BrowserRouter>
  </Provider>
</App>
`;

exports[`app shallows 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        key="/"
        path="/"
      >
        <Home />
      </Route>
      <Route
        exact={false}
        key="*"
        path="*"
      >
        <NotFound />
      </Route>
    </Switch>
  </BrowserRouter>
</Provider>
`;
